src/js/Snackbars/Snackbar.js

```diff
@@ -144,6 +155,7 @@ export default class Snackbar extends PureComponent {
     let { id } = this.props;
 
     let Component = 'p';
+    let closeButton;
```

Comment:
Sorry for taking so long to get back to you.. How about doing:

```
let childrenLeft;
let childrenRight;

...

if (action) {
  Component = 'section';
  text = <p className="md-snackbar--toast md-snackbar--action">{text}</p>;

  let btnProps = {
    flat: true,
    onClick: this._handleClick,
    label: action,
    secondary: true,
    className: 'md-btn--snackbar',
  };

  if (typeof action !== 'string') {
    const { childrenLeft: left, childrenRight: right, ...actionProps } = action;
    childrenLeft = left;
    childrenRight = right;

    btnProps = { ...btnProps, ...actionProps, ...{ className: cn(btnProps.className, action.className), onClick: this._handleClick };
    }
  }

...

{childrenLeft}
{text}
{action}
{childrenRight}
```

instead? This would give a lot more control than just placing a button, but it wouldn't automatically bind the close action to the `onClick`.

Let me know if that makes sense..